includes/classes/Indexable/Comment/QueryIntegration.php

```diff
@@ -0,0 +1,336 @@
+		 *                     site id (int or string), or an array of site ids (array).
+		 */
+		$scope = apply_filters( 'ep_comment_search_scope', $scope );
+
+		if ( ! defined( 'EP_IS_NETWORK' ) || ! EP_IS_NETWORK ) {
+			$scope = 'current';
+		}
+
+		$this->index = null;
+
+		if ( 'all' === $scope ) {
+			$this->index = $this->indexable->get_network_alias();
+		} elseif ( is_numeric( $scope ) ) {
+			$this->index = $this->indexable->get_index_name( (int) $scope );
+		} elseif ( is_array( $scope ) ) {
+			$this->index = [];
+
+			foreach ( $scope as $site_id ) {
+				$this->index[] = $this->indexable->get_index_name( $site_id );
+			}
+
+			$this->index = implode( ',', $this->index );
+		}
+
+		$ep_query = $this->indexable->query_es( $formatted_args, $query->query_vars, $this->index );
+
+		if ( false === $ep_query ) {
+			$query->elasticsearch_success = false;
+			return $results;
+		}
+
+		if ( ! empty( $query->query_vars['count'] ) ) {
+			return count( $ep_query['documents'] );
+		}
+
+		$query->found_comments        = $ep_query['found_documents'];
+		$query->elasticsearch_success = true;
+
+		// Determine how we should format the results from ES based on the fields parameter.
+		$fields = $query->query_vars['fields'];
+
+		switch ( $fields ) {
+			case 'count':
+				$new_comments = count( $ep_query['documents'] );
+				break;
+
+			case 'ids':
+				$new_comments = $this->format_hits_as_ids( $ep_query['documents'], $new_comments );
+				break;
+
+			default:
+				$new_comments = $this->format_hits_as_comments( $ep_query['documents'], $new_comments, $query->query_vars );
+				break;
+		}
+
+		return $new_comments;
+	}
+
+	/**
+	 * Format the ES hits/results as comments objects.
+	 *
+	 * @param  array $comments The comments that should be formatted.
+	 * @param  array $new_comments Array of comments from cache.
+	 * @param  array $query_vars Query variables.
+	 * @since  3.1
+	 * @return array
+	 */
+	protected function format_hits_as_comments( $comments, $new_comments, $query_vars ) {
+		$hierarchical = $query_vars['hierarchical'] ?? false;
+
+		foreach ( $comments as $comment_array ) {
+			$comment = new \stdClass();
+
+			$comment->ID      = $comment_array['comment_ID'];
+			$comment->site_id = get_current_blog_id();
+
+			if ( ! empty( $comment_array['site_id'] ) ) {
+				$comment->site_id = $comment_array['site_id'];
+			}
+
+			$comment_return_args = apply_filters(
+				'ep_search_comment_return_args',
+				[
+					'comment_ID',
+					'comment_post_ID',
+					'comment_author',
+					'comment_author_email',
+					'comment_author_url',
+					'comment_author_IP',
+					'comment_date',
+					'comment_date_gmt',
+					'comment_content',
+					'comment_karma',
+					'comment_approved',
+					'comment_agent',
+					'comment_type',
+					'comment_parent',
+					'user_id',
+				]
+			);
+
+			foreach ( $comment_return_args as $key ) {
+				if ( isset( $comment_array[ $key ] ) ) {
+					$comment->$key = $comment_array[ $key ];
+				}
+			}
```

Comment:
I think it's important to return `\WP_Comment` objects rather than plain ones, as it's quite possible the consuming code would check the object type.
```suggestion
			$comment = new \WP_Comment( (object) $comment_array );

			if ( ! empty( $comment_array['site_id'] ) ) {
				$comment->site_id = $comment_array['site_id'];
			} else {
				$comment->site_id = get_current_blog_id();
			}
```